mifosng-android/src/main/java/com/mifos/mifosxdroid/online/sharingaccount/SharingAccountFragment.java

```diff
@@ -0,0 +1,421 @@
+
+        return rootView;
+    }
+
+    @OnCheckedChanged(R.id.cb_allow_dividend_calculation)
+    public void onClickDividendAllowedCheckBox() {
+        isAllowDividend = cbDividendAllowed.isChecked();
+    }
+
+    @OnClick(R.id.img_add)
+    public void addCharges() {
+        showChargesLayout();
+    }
+
+    public void showChargesLayout() {
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
+        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(
+                recyclerSheets.getContext(), layoutManager.getOrientation());
+        recyclerSheets.setLayoutManager(layoutManager);
+        recyclerSheets.addItemDecoration(dividerItemDecoration);
+        recyclerSheets.setAdapter(shareChargesAdapter);
+        shareChargesAdapter.addChargeOptionList(chargeOptionSelected);
+        shareChargesAdapter.notifyDataSetChanged();
+    }
+
+    public void inflateSharingSpinners() {
+
+        mSharingProductsAdapter = new ArrayAdapter<>(getActivity(),
+                android.R.layout.simple_spinner_item, mListSharingProductsNames);
+        mSharingProductsAdapter
+                .setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spProduct.setAdapter(mSharingProductsAdapter);
+        spProduct.setOnItemSelectedListener(this);
+
+        mChargesAdapter = new ArrayAdapter<>(getActivity(), android.R.layout
+                .simple_spinner_item, mChargesNames);
+        mChargesAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spCharges.setAdapter(mChargesAdapter);
+        spCharges.setOnItemSelectedListener(this);
+
+        mLockinPeriodAdapter = new ArrayAdapter<>(getActivity(), android.R.layout
+                .simple_spinner_item, mListLockinPeriodOptions);
+        mLockinPeriodAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spLockinPeriod.setAdapter(mLockinPeriodAdapter);
+        spLockinPeriod.setOnItemSelectedListener(this);
+
+        mMinActivePeriodAdapter = new ArrayAdapter<>(getActivity(), android.R.layout
+                .simple_spinner_item, mListMinActivePeriodOptions);
+        mMinActivePeriodAdapter.setDropDownViewResource(
+                android.R.layout.simple_spinner_dropdown_item);
+        spMinActivePeriod.setAdapter(mMinActivePeriodAdapter);
+        spMinActivePeriod.setOnItemSelectedListener(this);
+
+        mSavingsAccountAdapter = new ArrayAdapter<>(getActivity(), android.R.layout
+                .simple_spinner_item, mListSavingsAccountOptions);
+        mSavingsAccountAdapter.setDropDownViewResource(
+                android.R.layout.simple_spinner_dropdown_item);
+        spSavingsAccount.setAdapter(mSavingsAccountAdapter);
+        spSavingsAccount.setOnItemSelectedListener(this);
+    }
+
+    @OnClick(R.id.btn_submit)
+    void submitSharingAccount() {
+        if (Network.isOnline(getContext())) {
```

Comment:
I don't think we should use `getContext()` we should use `getActivity()`. what do you say @droidchef

Reply:
I have changed it to getAcitivity().